test/net/http/test_httpresponse.rb

```diff
@@ -744,6 +744,46 @@ def test_inspect_response
     assert_equal '#<Net::HTTPUnknownResponse ??? test response readbody=true>', res.inspect
   end
 
+  def test_deconstruct_keys
+    res = Net::HTTPOK.new('1.1', '200', 'OK')
+    res.body = 'test body'
+    res['content-type'] = 'text/plain'
+
+    keys = res.deconstruct_keys(nil)
+    assert_equal '200', keys[:code]
+    assert_equal 'OK', keys[:message]
+    assert_equal '1.1', keys[:http_version]
+    assert_equal 'test body', keys[:body]
+    assert_equal 'text/plain', keys[:content_type]
+  end
+
+  def test_deconstruct_keys_with_specific_keys
+    res = Net::HTTPOK.new('1.1', '200', 'OK')
+    res.body = 'test body'
+
+    keys = res.deconstruct_keys([:code, :message])
+    assert_equal({code: '200', message: 'OK'}, keys)
+  end
+
+  def test_pattern_matching
+    res = Net::HTTPOK.new('1.1', '200', 'OK')
+    res['content-type'] = 'application/json'
+
+    begin
+      matched = instance_eval <<~RUBY, __FILE__, __LINE__ + 1
```

Comment:
This was a similar trick as to what we had used around pattern matching for the 2.6 to 2.7 push, so I'm repeating it here. The other methods should function even without syntax bridges like this.